### PR TITLE
Use standard Spark DataFrameReader to read back partitioned.

### PR DESCRIPTION
**Pull request by tpoterba**
Deleted all the nonsense.

### CODE REVIEW

**Comment by tpoterba** [comment]
Fixes #621

**Comment by tpoterba** [comment]
Fixes #667

File: src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala [hunk 1]
```diff
@@ -346,8 +358,14 @@ class VariantSampleMatrix[T](val metadata: VariantMetadata,
 
   def coalesce(k: Int): VariantSampleMatrix[T] = {
     val n = rdd.partitions.length
-    if (k >= n)
+    if (k >= n) {
+      warn(
+        s"""cannot coalesce to a larger number of partitions:
+            |  Dataset contains ${ rdd.partitions.size } partitions, requested $k partitions
+            |  In order to run with more partitions, use the --blocksize global option and reimport.
+         """.stripMargin)
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala]: Move this to the coalesce command.

Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala]: I didn't even realize this was here -- can you put this in your coalesce PR?

File: src/test/scala/org/broadinstitute/hail/utils/OrderedRDDSuite.scala [hunk 2]
```diff
@@ -134,7 +135,18 @@ class OrderedRDDSuite extends SparkSuite {
...
+        fname match {
+          case partRegex(i) => i.toInt
+          case _ => throw new PathIOException(s"invalid parquet file `$fname'")
+        }
+      }
+
+      val parquetFiles = hadoopGlobAll(Array(s"""$tmpRdd/rdd.parquet/*.parquet"""), hadoopConf)
+        .sortBy(getPartNumber)
+
+      val rddReadBack = sqlContext.read.parquet(parquetFiles: _*)
```
Comment (cseed) [inline src/test/scala/org/broadinstitute/hail/utils/OrderedRDDSuite.scala]: I'd abstract this as `RichSQLContext.readParquetPartitioned` or something.

**Comment by tpoterba** [comment]
hmm, don't merge yet -- I'm going to abstract the parquet search and sort code to a utility function

**Comment by cseed** [comment]
Two minor comments.  So good.

**Pull request merged**